K2Bridge/Factories/BucketFactory.cs

```diff
@@ -46,7 +51,55 @@ public static DateHistogramBucket CreateDateHistogramBucketFromDataRow(DataRow r
                     continue;
                 }
 
-                dhb.Aggs[clmn.ColumnName] = new System.Collections.Generic.List<double>() { Convert.ToDouble(row[clmn.ColumnName]) };
+                var columnNameInfo = clmn.ColumnName.Split('%');
+
+                if (columnNameInfo.Length > 1)
+                {
+                    // key%metric%value1%value2%keyed
+                    var key = columnNameInfo[0];
+                    var metric = columnNameInfo[1];
+
+                    // extract the percentiles values: from the second to the last-1
+                    var queryValues = columnNameInfo[2..^1];
+
+                    // extract the boolean: last item of the pattern elements array
+                    var keyed = bool.Parse(columnNameInfo[^1]);
+
+                    if (metric == "percentile")
+                    {
+                        dhb.Aggs[key] = new Dictionary<string, object>();
+                        var returnValues = new Dictionary<string, double>();
+
+                        var percentileValues = (JArray)row[clmn.ColumnName];
+
+                        for (var index = 0; index < queryValues.Length; index++)
+                        {
+                            var name = queryValues[index];
+                            var value = percentileValues[index].ToString();
+                            returnValues.Add(name, double.Parse(value, CultureInfo.InvariantCulture));
+                        }
+
+                        if (keyed)
+                        {
+                            // keyed ====> Dictionary<string, double>
+                            dhb.Aggs[key].Add("values", returnValues);
+                        }
+                        else
+                        {
+                            // not keyed ====> List<object(double,double)>
+                            dhb.Aggs[key].Add("values", returnValues.ToDictionary(item => double.Parse(item.Key, CultureInfo.InvariantCulture), item => item.Value).ToList());
+                        }
+                    }
+                }
+                else
+                {
+                    var columnName = clmn.ColumnName;
+                    logger.LogTrace("Defining the value for {columnName}", columnName);
+
+                    dhb.Aggs[columnName] = new Dictionary<string, object>() {
+                        { "value", Convert.ToDouble(row[columnName]) },
```

Comment:
double.Parse ?

Reply:
I changed it to double.Parse but I need the ToString():

`{ "value", double.Parse(row[columnName].ToString(), CultureInfo.InvariantCulture) },`

Reply:
I've ran the unit tests, and I think we never get to this flow, can you check?

Reply:
This is not a percentiles branch, so maybe no test is covering it